packages/sdk/src/evm/constants/urls.ts

```diff
@@ -44,7 +44,7 @@ export function getChainProvider(
     options.supportedChains = [
       // @ts-expect-error - we know this is a chain and it will work to build the map
       network,
-      ...options.supportedChains.filter((c) => c.chainId !== network.chainId),
+      ...options.supportedChains.filter((c) => c.chainId === network.chainId),
```

Comment:
This change doesn't make sense to me. Can you explain the logic?

Reply:
with `!==` the `supportedChains ` property was not having the user provided RPC Override information and thus not working.

Changing this logic to be in-sync with [this](https://github.com/thirdweb-dev/js/pull/1746/files#diff-3c93eef5707ade0c30c25d6dd3ffd8bf75095e1d4d761decf5d9b4b7566607bbR890), thus updating it `===` gets the `supportChains` property value to be same, which is then used in the `buildDefaultMap()`

`buildDefaultMap()` implementation

```
function buildDefaultMap(options: SDKOptionsOutput) {
  return options.supportedChains.reduce(
    (previousValue, currentValue) => {
      previousValue[currentValue.chainId] = currentValue;
      return previousValue;
    },
    {} as Record<number, ChainInfo>,
  );
}
```

Since there was just one value (object) in `supportedChains` property, the above was just using the SDK provided value and not the user provided one.

Reply:
I think the root cause is that we're using an array here and then reducing it as an object so any duplicates will get overriden.

I think this will work for now since the SDK assumes a single chain ATM, so this change is ok, but this logic is still fragile.

Will look into a proper fix for this

Reply:
Your case is special since you're passing a network slug + custom supported chains, usually you would pass your custom chain directly as the network.

Let's merge this to unblock you and I'll do another pass at it

Reply:
Cool thanks

Reply:
Needs some unit tests badly

Reply:
Cool.

So can I merge this? or you will do a proper fix and then merge it?

Please let me know. Will need to update Engine with the new SDK for it to work :)

Reply:
okies. thank you will merge. 

yeah need UT

Reply:
> Your case is special since you're passing a network slug + custom supported chains, usually you would pass your custom chain directly as the network.

Ahh got it. Maybe I can update it on Engine side then and we do not need to touch SDK..